Core/include/Acts/Propagator/EigenStepper.hpp

```diff
@@ -79,16 +79,17 @@ class EigenStepper {
                    const parameters_t& par, NavigationDirection ndir = forward,
                    double ssize = std::numeric_limits<double>::max(),
                    double stolerance = s_onSurfaceTolerance)
-        : pos(par.position(gctx)),
-          dir(par.unitDirection()),
-          p(par.absoluteMomentum()),
-          q(par.charge()),
-          t(par.time()),
+        : q(par.charge()),
           navDir(ndir),
           stepSize(ndir * std::abs(ssize)),
           tolerance(stolerance),
           fieldCache(mctx),
           geoContext(gctx) {
+      pars.template segment<3>(eFreePos0) = par.position(gctx);
+      pars.template segment<3>(eFreeDir0) = par.unitDirection();
+      pars[eFreeTime] = par.time();
+      pars[eFreeQOverP] = par.charge() / par.absoluteMomentum();
```

Comment:
> We have to be a bit careful here in what we want to actually use this for: if this should track the curvature then this is fine as you automatically get zero curvature for neutral particles. This would correctly give you the right trajectory but you loose the momentum information.
> 
> In the track parameters, this parameter is either `q/p` or `1/p` for uncharged particles. If we take the same definition here, than we can store the existing q/p values from the parameters.
> 
> My personal take is that the q/p parameter in the internal free vector should represent the curvature-defining parameter, i.e. its zero for neutral parameters and different from the q/p parameter in the track parameters. Then the actual computations do not need to care whether we have a charged or neutral particle. But that would mean that we still need to track `q` and `p` separately and update `p` after every propagation step. This only really applies to the dense stepper anyways.
> 
> Btw. no such complication exists for the straight line stepper. There the internal q/p parameter should be the same as the q/p parameter in the track parameters.

Mmmm, if you ask neutral bound parameters to be converted into free bound parameters then are you expecting `parameters[eOverP]` to be 0?

Reply:
It depends. For the pure track parameters: no. For neutral track parameters the `q/p` parameter will hold `1/p` and will be non-zero for finite p. This is clearly outlined e.g. in the charge interpreter documentation in `EventData/Charge.hpp`.

Here, the `FreeVector` is a purely internal state that does **not** have to have the identical content as the `FreeVector` in the parameters. Only when we request the track parameters should it match the track parameters definition. I am saying that for the particular, internal use within the `EigenStepper` it might be beneficial to use a slightly different definition (as outlined above) for this parameter.

Reply:
That would however re-introduce a conversion in all conversion function, which now can conveniently be called with 
`freeToBound(...,state.pars,...)`, which I think it's not worth it.

The `FreeVector` will be a private member of the `State` and I think we can go ahead and chose the most convenient version for us, even if it has a similar convention as mentioned above. Besides, we still have an open question whether reordering the internal parameters would speed up the covariance transport (could not reproduce so easily now).

Reply:
> That would however re-introduce a conversion in all conversion function, which now can conveniently be called with
> `freeToBound(...,state.pars,...)`, which I think it's not worth it.

That is a good point. From that perspective it is much easier to use exactly the same definition as in the track parameters. However, than I would suggest to not compute it manually as `par.charge() / par.absoluteMomentum()` but to use the existing `q/p` value from the input parameters. The track parameters already take care of using either `q/p` or `1/p` consistently.

Reply:
Excellent point, will change that.

Reply:
... but only when we have fully introduced the FreeParameters, though, because at the moment, the constructor is templated on the `parameters_t` type, and hence I have no access to either `eFreeQoverP` or `eBoundQoverP`, I plan to split this into two constructors anyway because for the free parameters the constructor would just be a assignment operation.

Reply:
> ... but only when we have fully introduced the FreeParameters, though, because at the moment, the constructor is templated on the `parameters_t` type, and hence I have no access to either `eFreeQoverP` or `eBoundQoverP`, I plan to split this into two constructors anyway because for the free parameters the constructor would just be a assignment operation.

Which is exactly what I did in #86

Reply:
I see.

Reply:
I changed it and enabled for the moment only construction from `BoundParameters` as @FabianKlimpel has suggested to do.